### PR TITLE
Only create users if a key is defined

### LINKED ISSUE #36: Skip user when they have no client key
In this example the client_key is empty:
```
"chef_server": {
   "client_key": "",
   "type": [ "user"],
   "orgs": {
     "myorg": {
       "enabled": true,
       "admin": true
     }
   }
}
```

Which causes an error from the `chef-server-ctl` command to fail the chef run.

### PR DESCRIPTION
**Pull request by luckymike**
Users without keys won't be able to access anything, so it's pointless (and probably confusing) to create them. Fixes #36

### CODE REVIEW

**Comment by luckymike** [comment]
Note, this PR anticipates #68, which fixes the .kitchen.yml to work with the latest package naming.

File: recipes/client.rb [hunk 1]
```diff
@@ -86,38 +86,40 @@
...
-          command "chef-server-ctl user-create #{item['client']} #{first_name} #{last_name} #{email} #{item['password']} > /dev/null 2>&1"
-          not_if "chef-server-ctl user-list | grep '^#{item['client']}$'"
-        end
-        if(item['pub_key'])
-          execute "set user key: #{item['client']}" do
-            if node['chef-server'][:version].to_f >= 12.1
-              command "chef-server-ctl add-user-key #{item['client']} --public-key-path #{key_file} --key-name populator"
-            else
-              command "chef-server-ctl add-user-key #{item['client']} #{key_file} --key-name populator"
+          unless item['pub_key'].empty?
```
Comment (chrisroberts) [inline recipes/client.rb]:
```ruby
unless(item['pub_key'].to_s.empty?)
```

File: recipes/client.rb [hunk 2]
```diff
@@ -86,38 +86,40 @@
...
+            end
+            item['full_name'] = item.fetch('full_name', item['client'].capitalize)
+            first_name = item['full_name'].split(' ').first.capitalize
+            last_name = item['full_name'].split(' ').last.capitalize
+            email = item.fetch('email', "#{item['client']}@example.com")
+            execute "create user: #{item['client']}" do
+              command "chef-server-ctl user-create #{item['client']} #{first_name} #{last_name} #{email} #{item['password']} > /dev/null 2>&1"
+              not_if "chef-server-ctl user-list | grep '^#{item['client']}$'"
+            end
+            if(item['pub_key'])
```
Comment (chrisroberts) [inline recipes/client.rb]: This conditional can be removed as this will always be true here

File: recipes/client.rb (hunk 1 above)
Comment (luckymike) [inline recipes/client.rb]: Man, I clearly was staring at this too long b/c I spent about 20 minutes trying to think of how to handle non-strings :/ 👍

File: recipes/client.rb (hunk 2 above)
Comment (luckymike) [inline recipes/client.rb]: This is an excellent point. Fixing

**Comment by cwjohnston** [comment]
@luckymike looks like this needs rebase now

**Comment by luckymike** [comment]
Rebased and spot-checked by running all ubuntu-14 suites

**Pull request merged**